### PR TITLE
[whisper-cpp] Add version 1.7.2

### PR DESCRIPTION
**Pull request by MartinDelille**
### Summary

Adding version 1.7.2 to whisper-cpp. The metal option is now enabled by default according to the original repository.
Should I set a different default value for the metal option depending of the version ?

#### Details

---

- [ ] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [ ] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [ ] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by AbrilRBS** [comment]
> Should I set a different default value for the metal option depending of the version ?

Yes, I think that would be the best approach, which makes sure users of the old versions dont get unexpected changes on their end :)

File: recipes/whisper-cpp/all/conanfile.py [hunk 1]
```diff
@@ -73,29 +73,25 @@ def _compilers_minimum_version(self):
...
-
     @property
     def _is_openvino_option_available(self):
         return Version(self.version) >= "1.5.2"
 
     def config_options(self):
         if is_apple_os(self):
             del self.options.with_blas
+            if Version(self.version) < "1.7.0":
+                self.default_options.metal = False
```
Comment (AbrilRBS) [inline recipes/whisper-cpp/all/conanfile.py:84]:
```suggestion
                self.options.metal = False
```

**Comment by MartinDelille** [comment]
I realized the PR and the first commit was improperly named. I fixed the PR name but shall I rename the first commit and force push or the commit details will be squashed when merged ?

File: recipes/whisper-cpp/all/conanfile.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def config_options(self):
         if is_apple_os(self):
             del self.options.with_blas
             if Version(self.version) < "1.7.0":
-                self.default_options.metal = False
+                self.options.metal = False
```
[the remark this answers is not in the document]
Comment (AbrilRBS) [reply recipes/whisper-cpp/all/conanfile.py:84]: So the option: Did not exist before 1.5.4, and in that version was added as False by default, and from 1.7.0 onwards, it's set to True?

**Comment by AbrilRBS** [comment]
@MartinDelille the commits will be squashed before merge, no worries about that, but thanks for checking!

File: recipes/whisper-cpp/all/conanfile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MartinDelille) [reply recipes/whisper-cpp/all/conanfile.py:84]: Yes

[the remark this answers is not in the document]
Comment (MartinDelille) [reply recipes/whisper-cpp/all/conanfile.py:84]: I removed version before 1.5.4 so we don't need to handle the case where the option didn't exist yet.

**Review by czoido — DISMISSED** [review]

File: recipes/whisper-cpp/all/conanfile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AbrilRBS) [reply recipes/whisper-cpp/all/conanfile.py:84]: Note that this change does not force the option for consumers, it just sets a default based on the version before any value is propagated form the consumer.

Also, we saw that in 1.5.2 this also was set to True by default https://github.com/ggerganov/whisper.cpp/blob/v1.5.2/CMakeLists.txt#L39 which seems like a bug in the recipe if this was ever false for those versions, so I think the ideal solution would be to set it to True always and delete the option if not on an apple os. wdyt @MartinDelille?

[the remark this answers is not in the document]
Comment (MartinDelille) [reply recipes/whisper-cpp/all/conanfile.py:84]: With this change introduced [here](018538bab7f6add0a90cff45e1bc549fc81cf251), people are not able to use metal before 1.7.0...

[the remark this answers is not in the document]
Comment (MartinDelille) [reply recipes/whisper-cpp/all/conanfile.py:84]: @czoido @AbrilRBS  Ok we just cross post. I agree to have it by default to `True` on apple os. Do you want me to remove this two lines ?

**Review by AbrilRBS — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**